hal/src/nRF52840/ble_hal.cpp

```diff
@@ -380,8 +381,9 @@ class BleObject::Observer {
     bool observerInitialized_;
     volatile bool isScanning_;                              /**< If it is scanning or not. */
     hal_ble_scan_params_t scanParams_;                      /**< BLE scanning parameters. */
+    bool scanUseCoded_ = false;                             /** If true, use coded phy scanning. Default false. */
     os_semaphore_t scanSemaphore_;                          /**< Semaphore to wait until the scan procedure completed. */
-    uint8_t scanReportBuff_[BLE_MAX_SCAN_REPORT_BUF_LEN];   /**< Buffer to hold the scanned report data. */
+    uint8_t scanReportBuff_[BLE_GAP_SCAN_BUFFER_EXTENDED_MAX_SUPPORTED];   /**< Buffer to hold the scanned report data. MUST BE EXTENDED_MAX TO USE CODED PHY */
```

Comment:
I would prefer to still use `BLE_MAX_SCAN_REPORT_BUF_LEN` here and change the `BLE_GAP_SCAN_BUFFER_MAX` to `BLE_GAP_SCAN_BUFFER_EXTENDED_MAX_SUPPORTED` in `ble_hal_impl.h`, so that we can avoid using the platform specific definitions in HAL driver directly.

Reply:
Agree, will make change.

Reply:
Done, please check my work.